overlord/devicestate/devicemgr.go

```diff
@@ -1686,6 +1688,18 @@ func (m *DeviceManager) StoreContextBackend() storecontext.Backend {
 	return storeContextBackend{m}
 }
 
+var timeutilIsNTPSynchronized = timeutil.IsNTPSynchronized
+
+func (m *DeviceManager) ntpSyncedOrWaitedLongerThan(maxWait time.Duration) bool {
+	if m.ntpSynchronized || time.Now().After(startTime.Add(maxWait)) {
+		return true
+	}
+
+	// ignore any errors that occurred while checking for ntp
+	m.ntpSynchronized, _ = timeutilIsNTPSynchronized()
```

Comment:
This should probably return  "false, ErrNoTimedate1" (or similar name) and then we make the policy decision to tread this as "is-synced" here instead of in the lower layer.